teamcity/pool_data_source.go

```diff
@@ -96,15 +102,29 @@ func (d *poolDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 
 	if pool.Size == nil {
 		state = models.PoolDataModel{
-			Name: types.StringValue(string(pool.Name)),
-			Size: basetypes.NewInt64Null(),
-			Id:   types.Int64Value(int64(*(pool.Id))),
+			Name:     types.StringValue(string(pool.Name)),
+			Size:     basetypes.NewInt64Null(),
+			Id:       types.Int64Value(int64(*(pool.Id))),
+			Projects: types.SetNull(types.StringType),
 		}
 	} else {
 		state = models.PoolDataModel{
-			Name: types.StringValue(string(pool.Name)),
-			Size: types.Int64Value(int64(*(pool.Size))),
-			Id:   types.Int64Value(int64(*(pool.Id))),
+			Name:     types.StringValue(string(pool.Name)),
+			Size:     types.Int64Value(int64(*(pool.Size))),
+			Id:       types.Int64Value(int64(*(pool.Id))),
+			Projects: types.SetNull(types.StringType),
+		}
+	}
```

Comment:
[minor] I think if we only set size to a variable in if here we could remove some duplication

Something like
```
var size = basetypes.NewInt64Null()
    if pool.Size != nil {
        size = types.Int64Value(int64(*(pool.Size)))
	}
	
    state = models.PoolDataModel{
        Name:     types.StringValue(string(pool.Name)),
        Size:     size,
        Id:       types.Int64Value(int64(*(pool.Id))),
        Projects: types.SetNull(types.StringType),
    }

Reply:
You're right, done!